src/components/icon/icon-registry.ts

```diff
@@ -0,0 +1,370 @@
+import {Injectable} from 'angular2/core';
+import {BaseException} from 'angular2/src/facade/exceptions';
+import {Http} from 'angular2/http';
+import {Observable} from 'rxjs/Rx';
+
+
+/** Exception thrown when attmepting to load an icon with a name that cannot be found. */
+export class MdIconNameNotFoundException extends BaseException {
+  constructor(iconName: string) {
+      super(`Unable to find icon with the name "${name}"`);
+  }
+}
+
+/**
+ * Exception thrown when attmepting to load SVG content that does not contain the expected
+ * <svg> tag.
+ */
+export class MdIconSvgTagNotFoundException extends BaseException {
+  constructor() {
+      super('<svg> tag not found');
+  }
+}
+
+/**
+  * Configuration for an icon, including the URL and possibly the cached SVG element.
+  * @internal
+  */
+class SvgIconConfig {
+  svgElement: SVGElement = null;
+  constructor(public url: string) {
+  }
+}
+
+/**
+ * Service to register and display icons used by the <md-icon> component.
+ * - Registers icon URLs by namespace and name.
+ * - Registers icon set URLs by namespace.
+ * - Registers aliases for CSS classes, for use with icon fonts.
+ * - Loads icons from URLs and extracts individual icons from icon sets.
+ */
+@Injectable()
+export class MdIconRegistry {
+  /**
+   * URLs and cached SVG elements for individual icons. Keys are of the format "[namespace]:[icon]".
+   */
+  private _svgIconConfigs = new Map<string, SvgIconConfig>();
+
+  /**
+   * SvgIconConfig objects and cached SVG elements for icon sets, keyed by namespace.
+   * Multiple icon sets can be registered under the same namespace.
+   */
+  private _iconSetConfigs = new Map<string, SvgIconConfig[]>();
+
+  /**
+   * Cache for icons loaded by direct URLs.
+   */
+  private _cachedIconsByUrl = new Map<string, SVGElement>();
+
+  /**
+   * In-progress icon fetches. Used to coalesce multiple requests to the same URL.
+   */
+  private _inProgressUrlFetches = new Map<string, Observable<string>>();
+
+  /**
+   * Map from font identifiers to their CSS class names. Used for icon fonts.
```

Comment:
You could have a custom icon font that uses ligatures and still set an alias for its CSS class and use it like ```<md-icon fontSet="myfont">home</md-icon>```, right?